pkg/plugin/test-keeper/metrics_test.go

```diff
@@ -0,0 +1,189 @@
+package testkeeper_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/prometheus/client_golang/prometheus"
+	. "github.com/arquillian/ike-prow-plugins/pkg/internal/test"
+	"gopkg.in/h2non/gock.v1"
+	"fmt"
+	"github.com/arquillian/ike-prow-plugins/pkg/github"
+	"github.com/arquillian/ike-prow-plugins/pkg/log"
+	dto "github.com/prometheus/client_model/go"
+	"strings"
+	"github.com/arquillian/ike-prow-plugins/pkg/utils"
+	"github.com/arquillian/ike-prow-plugins/pkg/github/service"
+	"github.com/arquillian/ike-prow-plugins/pkg/plugin/test-keeper"
+	"errors"
+)
+
+var _ = Describe("TestKeeper Metrics", func() {
+
+	var handler *testkeeper.GitHubTestEventsHandler
+	log := log.NewTestLogger()
+	toBe := func(status, description, context, detailsLink string) SoftMatcher {
+		return SoftlySatisfyAll(
+			HaveState(status),
+			HaveDescription(description),
+			HaveContext(context),
+			HaveTargetURL(fmt.Sprintf("%s/%s/%s.html", docStatusRoot, strings.ToLower(status), detailsLink)),
+		)
+	}
+
+	toHaveBodyWithWholePluginsComment := SoftlySatisfyAll(
+		HaveBodyThatContains(fmt.Sprintf(ghservice.PluginTitleTemplate, testkeeper.ProwPluginName)),
+		HaveBodyThatContains("@bartoszmajsak"),
+	)
+
+	BeforeEach(func() {
+		defer gock.OffAll()
+		handler = &testkeeper.GitHubTestEventsHandler{Client: NewDefaultGitHubClient(), BotName: botName}
+		testkeeper.RegisterMetrics()
+	})
+
+	AfterEach(func() {
+		testkeeper.UnRegisterMetricsAndReset()
+		EnsureGockRequestsHaveBeenMatched()
+	})
+
+	It("should report pull requests with /ok-without-tests in histogram", func() {
+		//given
+		expectedBound := []float64{1, 5, 25, 125, 625, 3125}
+		expectedCnt := []uint64{0, 1, 1, 1, 1, 1, 1}
+
+		// given
+		NonExistingRawGitHubFiles("test-keeper.yml", "test-keeper.yaml")
+
+		gock.New("https://api.github.com").
+			Get("/repos/" + repositoryName + "/pulls/1").
+			Reply(200).
+			Body(FromFile("test_fixtures/github_calls/prs/without_tests/pr_details.json"))
+
+		gock.New("https://api.github.com").
+			Get("/repos/" + repositoryName + "/collaborators/bartoszmajsak/permission").
+			Reply(200).
+			Body(FromFile("test_fixtures/github_calls/collaborators_repo-admin_permission.json"))
+
+		gock.New("https://api.github.com").
+			Get("/repos/" + repositoryName + "/pulls/1/reviews").
+			Reply(200).
+			BodyString(`[]`)
+
+		toHaveEnforcedSuccessState := SoftlySatisfyAll(
+			HaveState(github.StatusSuccess),
+			HaveDescription(fmt.Sprintf(testkeeper.ApprovedByMessage, "bartoszmajsak")),
+		)
+
+		gock.New("https://api.github.com").
+			Post("/repos/" + repositoryName + "/statuses").
+			SetMatcher(ExpectPayload(toHaveEnforcedSuccessState)).
+			Reply(201) // This way we implicitly verify that call happened after `HandleEvent` call
+
+		statusPayload := LoadFromFile("test_fixtures/github_calls/prs/without_tests/skip_comment_by_admin.json")
+
+		// when
+		err := handler.HandleEvent(log, github.IssueComment, statusPayload)
```

Comment:
try to invoke the handler several times with different type of the payload. This would give me more information what is expected from the logic